Federate edge image list

acosferreira opened the pull request:
Initial version for the parity between image builder and edge image list

acosferreira commented:
FYI @mgold1234 @matoval @loadtheaccumulator This is the initial setup for the parity.
@croissanne The feature flag and layout approach is under the other pr's opened by Michal, please let us know if you need any changes on our side

croissanne commented on src/Components/ImagesTable/ImagesTable.js:
```diff
@@ -1,5 +1,8 @@
 import React, { useEffect, useState } from 'react';
 
+import AsyncComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+
```
no newline here

croissanne commented on src/Components/ImagesTable/ImagesTable.js:
```diff
@@ -26,7 +29,9 @@ import {
 import PropTypes from 'prop-types';
 import { useDispatch, useSelector } from 'react-redux';
-import { Link, useNavigate } from 'react-router-dom';
+
```
no newline here, should be 1 import block

croissanne reviewed (changes requested):
overall lgtm but can you clean up the git history a bit?

1. no merge commits, we strictly rebase
2. please run the linter on each commit individually, this avoids noise in later commits
3. read https://www.osbuild.org/guides/developer-guide/workflow.html to get more information about the format of commits and PRs

croissanne commented on src/Components/ImagesTable/ImagesTable.js:
```diff
@@ -154,6 +158,37 @@ const ImagesTable = () => {
   const itemsStartInclusive = (page - 1) * perPage;
   const itemsEndExclusive = itemsStartInclusive + perPage;
 
+  const notificationProp = {
```
https://github.com/RedHatInsights/image-builder-frontend/pull/1160 seems to add a copy of this, maybe just put this prop in `src/Utilities/edge.js` directly ? When 1160 gets rebased we won't have duplicate code then.

acosferreira replied:
@croissanne the src/Utilities/edge.js was added on #1160 can we do the clean up when rebase?

croissanne commented on src/Components/ImagesTable/ImagesTable.js:
```diff
@@ -24,10 +25,11 @@ import {
 import { useDispatch, useSelector } from 'react-redux';
-import { Link, useNavigate } from 'react-router-dom';
-
+import { Link, useNavigate, useLocation } from 'react-router-dom';
```
Ah sorry, I was wrong, the linter complains. There's 1 newline needed between the external and internal imports.

croissanne reviewed (changes requested):
Thanks! 2 small comments otherwise lgtm.

acosferreira commented:
/retest

mgold1234 reviewed (commented):
@croissanne those changes appear in my pr, maybe let's merge my pr first with the correct tab and location of the button and then continue to add there the implementation of the federation module that Ana did?

acosferreira commented:
@croissanne I've addressed some of your comments, and as @mgold1234 mentioned, the other layout changes are on ther pr.
Let us know the better way to proceed

acosferreira commented:
![image](https://github.com/RedHatInsights/image-builder-frontend/assets/5039367/d7ed0b3d-ace8-460c-9ece-d4cbc1a268cd)

acosferreira commented:
![image](https://github.com/RedHatInsights/image-builder-frontend/assets/5039367/663cebea-1731-4390-84ed-b0861628fe9b)

croissanne commented:
> ![image](https://github.com/RedHatInsights/image-builder-frontend/assets/5039367/663cebea-1731-4390-84ed-b0861628fe9b)
> 

Hmm, that still doesn't have the white background for the tab headers.

Also I think every PR should make sense and work  as a unit. So we can either rebase the other on top of this. Or we can merge them one by one, but both need to be green of course.

acosferreira commented:
@mgold1234 lets rebase our prs and get in one then we can have the complete flow working.
@croissanne the tests that is broken is related to the tab changes we are looking to fix that based on the feature flag.

acosferreira commented:
Closing this pr, once the changes were rebased on #1149
